tests/system/Log/ChromeLoggerHandlerTest.php

```diff
@@ -56,23 +55,26 @@ public function testSetDateFormat()
 		$this->assertObjectHasAttribute('dateFormat', $logger);
 	}
 
-	//--------------------------------------------------------------------
 
-	public function testObjectMessage()
+	public function testChromeLoggerHeaderSent()
 	{
-		$config                                                              = new LoggerConfig();
+		Services::injectMock('response', new MockResponse(new App()));
+		$response = service('response');
+
+		$config = new LoggerConfig();
+
 		$config->handlers['CodeIgniter\Log\Handlers\TestHandler']['handles'] = ['critical'];
 
-		$logger            = new MockChromeLogger($config->handlers['CodeIgniter\Log\Handlers\TestHandler']);
+		$logger = new ChromeLoggerHandler($config->handlers['CodeIgniter\Log\Handlers\TestHandler']);
+
 		$data              = new \stdClass();
 		$data->code        = 123;
 		$data->explanation = "That's no moon, it's a pumpkin";
 		$result            = $logger->setDateFormat('F j, Y');
 
-		$logger->handle('debug', $data);
-		$peek = $logger->peekaboo();
+		$logger->handle('warning', $data);
 
-		$this->assertEquals($data->explanation, $peek[0]['explanation']);
+		$this->assertTrue($response->hasHeader('x-chromelogger-data'));
```

Comment:
Maybe I'm mis-reading something here, but this should actually be false, shouldn't it? If `TestHandler` is only set to log `critical` errors, and we tell the logger to handle a `warning`, then TestHandler (and therefor `ChromeLoggerHandler`) shouldn't send any data, right?

Reply:
The decision to actually log (or not) is made in ``\CodeIgniter\Log\Logger::log()`` and despite the object being called ``$logger`` it is not an instance of ``Logger`` it is an instance of ``ChromeLoggerHandler``.  The severity level of the message has no bearing on this test.

Reply:
Doh! You're right. Thanks for clarifying. I'm good with this then.